appserver/concurrent/concurrent-impl/src/main/java/org/glassfish/concurrent/runtime/ContextSetupProviderImpl.java

```diff
@@ -239,12 +239,16 @@ private void startConcurrentContextSpan(ComponentInvocation invocation, Invocati
 
         // Check for propagated span
         if (handle.getSpanContextMap() != null) {
+            @SuppressWarnings("unchecked")
             SpanContext spanContext = tracer.extract(Format.Builtin.TEXT_MAP, new MapToTextMap(handle.getSpanContextMap()));
             builder.asChildOf(spanContext);
 
             // Check for the presence of a propagated parent operation name
             try {
-                String operationName = ((RequestTraceSpanContext) spanContext).getBaggageItems().get("operation.name");
+                String operationName = null;
+                if (spanContext != null) {
```

Comment:
Shouldn't this check be further up? Before `builder.asChildOf`? Otherwise you're registering this new Span as a child of `null`

In fact, this scenario seems odd to me.
All Spans should have a SpanContext and an operation name - if we're getting a null on either of them despite the fact that `handle.getSpanContextMap() != null` returned true it seems to me that there's a badly made Span somewhere, or that `extract` has broken

Reply:
There is definitely badly made span somewhere. However I am unable to reliably reproduce the source fault to be able to investigate. 
this fix is so I’m bad span doesn’t cause cascade failures.

Reply:
> Shouldn't this check be further up? Before builder.asChildOf? 

Not sure. Here the "old" code checks for `operation.name != null` and the new code
checks for `operation.name` that doesn't exist. Not much difference in functionality as far as I 
can see fo I think this fix is pragmatic and correct.